### PR TITLE
feat: Opentelemetry Support

### PR DESCRIPTION
**Pull request by lxfontes**
Sets up Otel exporters for logs/metrics/traces based on  Otel configuration from HostData. Related to #23 .

We configure opentelemetry metrics / logging / tracing as part of provider startup. We setup the base resource with:
- ServiceName: binary base name
- Service Instance Id: The provider key, from host data
Provider authors can call the usual Opentelemetry methods relay data. See key-value example.

Sample Report:
```
Resource attributes:
     -> service.instance.id: Str(test-http_kubernetes)
     -> service.name: Str(file____users_lucas_code_src_github_com_wasmcloud_http_kubernetes_build_http_kubernetes_par_gz)
ScopeSpans #0
ScopeSpans SchemaURL:
InstrumentationScope http-kubernetes
Span #0
    Trace ID       : d654cf4dc37e612d038ec507e8af9cd5
    Parent ID      :
    ID             : 18ea2eb49a0c2864
    Name           : ServeHTTP
    Kind           : Internal
    Start time     : 2024-08-05 15:23:44.19262 +0000 UTC
    End time       : 2024-08-05 15:23:44.199920667 +0000 UTC
    Status code    : Ok
    Status message :
	{"kind": "exporter", "data_type": "traces", "name": "debug"}
```

## Setup examples
Tracing:
```
import  "go.opentelemetry.io/otel"

tracer := otel.Tracer("my-tracer-name")

......

// Parent Span-Id will be inferred from 'ctx'
ctx, span := tracer.Start(ctx, "myFunc")

... do some work within span ...

span.End()
```

Metrics:
```
import  "go.opentelemetry.io/otel"

meter := otel.Meter("my-meter-name")
rollCnt, err = meter.Int64Counter("dice.rolls", metric.WithDescription("The number of rolls by roll value"), metric.WithUnit("{roll}"))
if err != nil {
  panic(err)
}

......

roll := 1 + rand.Intn(10)
rollValueAttr := attribute.Int("roll.value", roll)
rollCnt.Add(ctx, 1, metric.WithAttributes(rollValueAttr))
```

Logs:
```
import "go.opentelemetry.io/contrib/bridges/otelslog"

logger = otelslog.NewLogger("my-logger-name")

....

roll := 1 + rand.Intn(10)
logger.InfoContext(ctx, "Rolled dice", "result", roll)
```

### CODE REVIEW

File: observability.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	case OtelProtocolHTTP:
+		exporter, err = otlptracehttp.New(ctx, otlptracehttp.WithEndpointURL(endpoint))
+	default:
+		return nil, fmt.Errorf("unknown observability protocol '%s'", config.Protocol)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	spanLimits := trace.SpanLimits{
```
Comment (lxfontes) [inline observability.go:58]: copied from wasmcloud host https://github.com/wasmCloud/wasmCloud/blob/b3c6ccb77150cd2419efececb4956e8b249d3381/crates/tracing/src/traces.rs#L239

File: observability.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+const (
+	OtelMetricExportInterval = 1 * time.Minute
+	OtelTraceExportInterval  = 1 * time.Minute
+	OtelLogExportInterval    = 10 * time.Second
```
Comment (joonas) [inline observability.go:27]: I'm curious where these values come from, are they something that we've got set up somewhere already, or are they the defaults from upstream? i.e. how'd we land on these values?

Comment (lxfontes) [reply observability.go:27]: The default values are:
- logging: 1 second
- metrics: 60 seconds
- traces: 5 seconds

This is net new / not configured in wasmcloud host atm. 
IMO: Would rather not hardcode these. Either leave the defaults or take values from HostData.

**Review by joonas — APPROVED** [review]

File: observability.go (hunk 2 above)
Comment (joonas) [reply observability.go:27]: That seems totally reasonable

**Review by brooksmtownsend — APPROVED** [review]
This looks great! I just had two nits around the example & a comment, so approving and can address now or as a nit followup

**Comment by lxfontes** [comment]
@brooksmtownsend addressing PR comments 🎉 🚢

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**